reward/src/lib.rs

```diff
@@ -0,0 +1,532 @@
+    term::blank();
+    let mut table = term::Table::default();
+    table.push([
+        term::format::bold("Address"),
+        term::format::bold("Reward"),
+        term::format::bold("ENS"),
+    ]);
+    let mut receivers = Vec::new();
+    let mut amounts = Vec::new();
+    for (ens, reward) in rewards.iter() {
+        let resolver: Result<PublicResolver<SignerMiddleware<_, _>>, _> =
+            rt.block_on(PublicResolver::get(ens, signer.clone()));
+
+        // if resolver doesn't exist, we just skip this one
+        if resolver.is_err() {
+            table.push([
+                term::format::italic(term::format::negative("Missing")),
+                term::format::tertiary(u256_to_amount(*reward, decimals)?),
+                term::format::secondary(ens),
+            ]);
+            continue;
+        }
+        let resolver = resolver.unwrap();
+
+        let address = rt
+            .block_on(resolver.address(ens))?
+            .ok_or_else(|| anyhow!("Couldn't get Address of ENS"))?;
+
+        receivers.push(address);
+        amounts.push(*reward);
+
+        table.push([
+            term::format::highlight(address),
+            term::format::tertiary(u256_to_amount(*reward, decimals)?),
+            term::format::secondary(ens),
+        ]);
+    }
+    table.render();
+    term::blank();
+
+    // exit now if this was a dry run
+    if options.dry {
+        return Ok(());
+    }
+
+    if term::confirm("Do you wish to proceed?") {
+        let sum = amounts
+            .iter()
+            .fold(Some(U256::from(0_u64)), |sum, amt| {
+                sum.map(|s| s.checked_add(*amt)).unwrap()
+            })
+            .ok_or_else(|| anyhow!("Failed to sum all transactions amounts"))?;
+
+        let token: ERC20<SignerMiddleware<_, _>> = ERC20::new(signer.clone(), address);
+        let spinner = term::spinner(&term::format::tertiary("Checking allowance..."));
+        let allowance = rt.block_on(token.get_allowance(signer.address(), *SUPERSEEDER_ADDRESS))?;
+        spinner.finish();
+
+        if allowance < sum {
+            term::blank();
+            term::info!(
+                "{}",
+                term::format::tertiary("Approving transaction amount for Superseeder...")
+            );
+            term::blank();
+
+            let call = token.approve(*SUPERSEEDER_ADDRESS, sum)?;
+            rt.block_on(term::ethereum::transaction(call))?;
+        }
+
+        // generate tx(s)
+        let superseeder: Superseeder<SignerMiddleware<_, _>> = Superseeder::new(signer);
+        let call = superseeder.send(address, receivers, amounts)?;
+        term::blank();
+        term::info!("{}", term::format::tertiary("Sending transaction..."));
+        term::blank();
+        rt.block_on(term::ethereum::transaction(call))?;
+
+        // tag HEAD as last successful payout
+        if head_sha1.is_some() {
+            let spinner = term::spinner(&term::format::tertiary("Tagging HEAD as `rad/reward`"));
```

Comment:
It's not `HEAD` right, it's whatever the `head_sha1` commit is?

Reply:
https://github.com/radicle-dev/radicle-cli/compare/f4007e459f8ce654f3ca58ce48dc1f9910838534..a664622f7600c99dc0622b776816712d0b049d45#diff-26d7e6053383a7793e880ae8b6ee5d2f7d7c23ccd5e88e43f44052374e9f0f58L392-R387